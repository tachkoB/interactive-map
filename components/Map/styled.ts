import styled from 'styled-components';
import { ComposableMap as Map } from 'react-simple-maps';

export const ComposableMap = styled(Map)`
  border: 2px solid black;
  width: 75%;
  height: 75%;
  background-color: #000080;
  border: 4px solid white;
  border-radius: 15px; ;
`;
