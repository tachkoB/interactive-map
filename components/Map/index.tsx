import { FC, PropsWithChildren } from 'react';
import { ZoomableGroup } from 'react-simple-maps';

// Styles
import { ComposableMap } from './styled';

interface Props extends PropsWithChildren {
  height: number;
  width: number;
}

/**
 * Renders a a zoomable, interactive map
 */
const MapChart: FC<Props> = ({ height, width, children }) => {
  return (
    <ComposableMap data-tip width={width} height={height}>
      <ZoomableGroup
        translateExtent={[
          [0, 0],
          [width, height],
        ]}
      >
        {children}
      </ZoomableGroup>
    </ComposableMap>
  );
};

export default MapChart;
