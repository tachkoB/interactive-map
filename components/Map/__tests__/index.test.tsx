import Map from '../index';

import { render } from '@testing-library/react';

describe('Map component', () => {
  it('should render the component', () => {
    const { container } = render(<Map height={200} width={200} />);

    expect(container).toMatchSnapshot();
  });

  it('should render the children', () => {
    const { getByText } = render(
      <Map height={200} width={200}>
        hello
      </Map>
    );

    expect(getByText('hello')).toBeInTheDocument();
  });
});
