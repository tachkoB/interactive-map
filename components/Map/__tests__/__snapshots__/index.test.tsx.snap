// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map component should render the component 1`] = `
<div>
  <svg
    class="rsm-svg sc-bczRLJ cIlMek"
    data-tip="true"
    viewBox="0 0 200 200"
  >
    <g>
      <rect
        fill="transparent"
        height="200"
        width="200"
      />
      <g
        class="rsm-zoomable-group undefined"
        transform="translate(0 0) scale(1)"
      />
    </g>
  </svg>
</div>
`;
